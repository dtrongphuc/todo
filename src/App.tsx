import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import LoginPage from 'pages/Login/LoginPage';
import HomePage from 'pages/HomePage';

function App() {
  return (
    <Router>
      <Switch>
        <Route exact path='/auth/login' component={LoginPage} />
        <Route path='/' component={HomePage} />
      </Switch>
    </Router>
  );
}

export default App;
