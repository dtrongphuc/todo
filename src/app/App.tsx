import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import LoginPage from './pages/LoginPage/LoginPage';
import AuthProvider from 'contexts/AuthProvider';
import MainLayout from 'layouts/MainLayout';
import TasksPage from './pages/TasksPage/TasksPage';
import SearchPage from './pages/SearchPage/SearchPage';
import PrivateRoute from './components/PrivateRoute/PrivateRoute';
import CompletedPage from './pages/CompletedPage/CompletedPage';
import ImportantPage from './pages/ImportantPage/ImportantPage';

function App() {
  return (
    <AuthProvider>
      <Router>
        <Switch>
          <Route exact path='/auth/login' component={LoginPage} />
          <PrivateRoute path='/'>
            <MainLayout>
              <PrivateRoute exact path='/' component={TasksPage} />
              <PrivateRoute exact path='/completed' component={CompletedPage} />
              <PrivateRoute exact path='/important' component={ImportantPage} />
              <PrivateRoute exact path='/search' component={SearchPage} />
            </MainLayout>
          </PrivateRoute>
        </Switch>
      </Router>
    </AuthProvider>
  );
}

export default App;
