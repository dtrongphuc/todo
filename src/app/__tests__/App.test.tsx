import App from 'app/App';
import { shallow } from 'enzyme';

describe('App component', () => {
  it('matches the snapshot', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
