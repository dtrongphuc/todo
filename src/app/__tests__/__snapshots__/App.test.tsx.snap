// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches the snapshot 1`] = `
<BrowserRouter>
  <AuthProvider
    provider={
      Object {
        "getIdentity": [Function],
        "login": [Function],
        "logout": [Function],
      }
    }
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/auth/login"
      />
      <PrivateRoute
        path="/"
      >
        <MainLayout>
          <PrivateRoute
            component={[Function]}
            exact={true}
            path="/"
          />
          <PrivateRoute
            component={[Function]}
            exact={true}
            path="/completed"
          />
          <PrivateRoute
            component={[Function]}
            exact={true}
            path="/important"
          />
          <PrivateRoute
            component={[Function]}
            exact={true}
            path="/search"
          />
        </MainLayout>
      </PrivateRoute>
    </Switch>
  </AuthProvider>
</BrowserRouter>
`;
